src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -95,12 +98,131 @@ public static IResourceBuilder<RedisResource> WithRedisCommander(this IResourceB
         }
     }
 
+    /// <summary>
+    /// Configures a container resource for Redis Insight which is pre-configured to connect to the <see cref="RedisResource"/> that this method is used on.
+    /// </summary>
+    /// <param name="builder">The <see cref="IResourceBuilder{T}"/> for the <see cref="RedisResource"/>.</param>
+    /// <param name="configureContainer">Configuration callback for Redis Insight container resource.</param>
+    /// <param name="containerName">Override the container name used for Redis Insight.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<RedisResource> WithRedisInsight(this IResourceBuilder<RedisResource> builder, Action<IResourceBuilder<RedisInsightResource>>? configureContainer = null, string? containerName = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        if (builder.ApplicationBuilder.Resources.OfType<RedisInsightResource>().SingleOrDefault() is { } existingRedisCommanderResource)
+        {
+            var builderForExistingResource = builder.ApplicationBuilder.CreateResourceBuilder(existingRedisCommanderResource);
+            configureContainer?.Invoke(builderForExistingResource);
+            return builder;
+        }
+        else
+        {
+            builder.ApplicationBuilder.Services.AddHttpClient();
+            containerName ??= $"{builder.Resource.Name}-insight";
+
+            var resource = new RedisInsightResource(containerName);
+            var resourceBuilder = builder.ApplicationBuilder.AddResource(resource)
+                                      .WithImage(RedisContainerImageTags.RedisInsightImage, RedisContainerImageTags.RedisInsightTag)
+                                      .WithImageRegistry(RedisContainerImageTags.RedisInsightRegistry)
+                                      .WithHttpEndpoint(targetPort: 5540, name: "http")
+                                      .ExcludeFromManifest();
+
+            builder.ApplicationBuilder.Eventing.Subscribe<AfterResourcesCreatedEvent>(async (e, ct) =>
```

Comment:
You don't want this event, this will run after all resources are finished, not just this one.

Reply:
@mitchdenny This is where it would be useful to have a resource healthy event.

Reply:
> You don't want this event, this will run after all resources are finished, not just this one.

@davidfowl I couldn't find any appropriate API for this one.

Reply:
I just realized that this needs to look at all resources so that’s fine.